### ISSUE TITLE
MetricCollection doesn't merge compute groups on forward

### ISSUE THREAD

**Issue by tsteffek** [Lightning-AI/metrics#1083]
## 🐛 Bug

So I'm not entirely sure whether this is a bug or intentional, in that case this would be a doc fix request.

As the title states, the `MetricCollection` will not merge compute groups on `forward()`. While `update()` merges the compute groups and calls update on all metrics, forward will only call forward on all metrics and ignore compute groups altogether.

This is contradictory to the example in the [docs](https://torchmetrics.readthedocs.io/en/stable/pages/overview.html#metriccollection), which shows an example using forward and claims this procedure would gain a speedup from compute groups.

### To Reproduce

Setup a `MetricCollection` with merge-able Metrics, call forward on it and observe no merging:

#### Code sample

```python
import torch, torchmetrics

m = torchmetrics.MetricCollection(torchmetrics.Accuracy(), torchmetrics.Recall())
# Binary inputs
binary_preds  = torch.tensor([0.6, 0.1, 0.9])
binary_target = torch.tensor([1, 0, 1])

m._groups
# Out[28]: {0: ['Accuracy'], 1: ['Recall']} - initially not merged, this is fine

m(binary_preds, binary_target)
# Out[31]: {'Accuracy': tensor(1.), 'Recall': tensor(1.)}
m._groups
# Out[32]: {0: ['Accuracy'], 1: ['Recall']} - still not merged, this is bad(?)
m.update(binary_preds, binary_target)
m._groups
# Out[34]: {0: ['Accuracy', 'Recall']} - update correctly merges the metrics
```

### Expected behavior

Merge Groups should be merged and used, I think, i.e. forward should behave as update in that regard. Alternatively I can see how, since forward and update behave differently (one returning the metric, the other not), compute groups might not be as effective here. In that case, the docs should be updated to not promise compute group speedups on forward calls, and ideally a warning describing this behavior should be added.

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): `0.9.1`, `pip`
- Python & PyTorch Version (e.g., 1.0):  `3.8.12` & `1.11.0`
- Any other relevant information such as OS (e.g., Linux): `FROM pytorch/pytorch:1.11.0-cuda11.3-cudnn8-devel`
